developer/bake_config.py

```diff
@@ -127,22 +141,41 @@ def bake_config(sg_connection, config_uri, target_path, do_zip=False):
     filesystem.ensure_folder_exists(bundle_cache_root)
     logger.info("Downloading and caching config...")
     config_descriptor.clone_cache(bundle_cache_root)
-    cache_apps(sg_connection, config_descriptor, bundle_cache_root)
 
-    # Now analyze what core the config needs and cache it if needed.
-    if config_descriptor.associated_core_descriptor:
-        logger.info("Config is specifying a custom core in config/core/core_api.yml.")
-        logger.info("This will be used when the config is executing.")
+    # If sparse_caching is True, we add app_store descriptor types to the skip
+    # list in order to keep our bundle cache small and let Desktop handle
+    # downloading app_store bundles separately.
+    should_skip_caching_callable = None
+    if sparse_caching:
         logger.info(
-            "Ensuring this core (%s) is cached..." % config_descriptor.associated_core_descriptor
-        )
-        associated_core_desc = create_descriptor(
-            sg_connection,
-            Descriptor.CORE,
-            config_descriptor.associated_core_descriptor,
-            bundle_cache_root_override=bundle_cache_root
+            "Performing sparse caching. Will not cache standard app_store bundles."
         )
-        associated_core_desc.ensure_local()
+        should_skip_caching_callable = _should_skip_caching_sparse
+
+    cache_apps(sg_connection, config_descriptor, bundle_cache_root, should_skip_caching_callable)
+
+    # Now analyze what core the config needs and cache it if needed.
+    core_descriptor = config_descriptor.associated_core_descriptor
+    if core_descriptor:
+        logger.info("Config defines a specific core in config/core/core_api.yml.")
+        logger.info("This will be used when the config is executing.")
+        if not sparse_caching or not _should_skip_caching_sparse(core_descriptor):
```

Comment:
A comment would be helpful here, something like `If sparse_caching is on, check if we need to cache tk-core or not`.